src/ol/view.js

```diff
@@ -256,10 +256,17 @@ ol.View.prototype.getCenter = function() {
 
 
 /**
+ * @param {Array.<number>=} opt_hints Destination array.
  * @return {Array.<number>} Hint.
  */
-ol.View.prototype.getHints = function() {
-  return this.hints_.slice();
+ol.View.prototype.getHints = function(opt_hints) {
```

Comment:
Not sure about this; `frameState.viewHints` is updated in (at least) the [animation functions](https://github.com/openlayers/ol3/blob/v3.15.1/src/ol/animation.js#L32)